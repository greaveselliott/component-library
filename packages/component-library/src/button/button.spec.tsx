import * as React from 'react';
import { shallow } from 'enzyme';
import { Button } from '.';

describe('Button', () => {
  it('renders as expected', () => {
    expect(
      shallow(<Button shape="round" variant="secondary" size="small" />)
    ).toMatchSnapshot();
  });

  it('renders a link as expected', () => {
    expect(
      shallow(<Button href="#" shape="default" variant="secondary" />)
    ).toMatchSnapshot();
  });

  it('renders a button elemement as default', () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  });

  it('renders an anchor tag if given a href', () => {
    expect(shallow(<Button href="#" />)).toMatchSnapshot();
  });

  it('is able to call a function on click', () => {
    const mockOnClickFunction = jest.fn();
    const button = shallow(<Button onClick={mockOnClickFunction} />);
    button.simulate('click');
    expect(mockOnClickFunction).toHaveBeenCalled();
  });

  it('can be disabled', () => {
    expect(shallow(<Button disabled />)).toMatchSnapshot();
  });

  it('can render children', () => {
    expect(shallow(<Button disabled>abc</Button>)).toMatchSnapshot();
  });
});
