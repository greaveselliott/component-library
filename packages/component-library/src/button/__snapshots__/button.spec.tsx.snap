// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be disabled 1`] = `
<button
  className="button button--primary button--default"
  disabled={true}
/>
`;

exports[`Button can render children 1`] = `
<button
  className="button button--primary button--default"
  disabled={true}
>
  abc
</button>
`;

exports[`Button renders a button elemement as default 1`] = `
<button
  className="button button--primary button--default"
/>
`;

exports[`Button renders a link as expected 1`] = `
<a
  className="button button--secondary button--default"
/>
`;

exports[`Button renders an anchor tag if given a href 1`] = `
<a
  className="button button--primary button--default"
/>
`;

exports[`Button renders as expected 1`] = `
<button
  className="button button--secondary button--round button--round--small"
/>
`;
